### PR TITLE
[NEW] Added push.token requests implementation

### PR DESCRIPTION
**Pull request by leonardoaramaki**
@RocketChat/android-sdk

### CODE REVIEW

File: core/src/main/kotlin/chat/rocket/core/internal/rest/Push.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import com.squareup.moshi.Types
+import okhttp3.MediaType
+import okhttp3.RequestBody
+
+suspend fun RocketChatClient.registerPushToken(token: String) {
+    val payload = PushRegistrationPayload(value = token)
+    val adapter = moshi.adapter(PushRegistrationPayload::class.java)
+
+    val payloadBody = adapter.toJson(payload)
+    val contentType = MediaType.parse("application/json; charset=utf-8")
```
Comment (rafaelks) [inline core/src/main/kotlin/chat/rocket/core/internal/rest/Push.kt]: I guess our contentType could be by default this one, right @luciofm & @leonardoaramaki?

File: core/src/main/kotlin/chat/rocket/core/internal/rest/Push.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import chat.rocket.core.model.PushToken
+import com.squareup.moshi.Types
+import okhttp3.MediaType
+import okhttp3.RequestBody
+
+suspend fun RocketChatClient.registerPushToken(token: String) {
+    val payload = PushRegistrationPayload(value = token)
+    val adapter = moshi.adapter(PushRegistrationPayload::class.java)
+
+    val payloadBody = adapter.toJson(payload)
```
Comment (rafaelks) [inline core/src/main/kotlin/chat/rocket/core/internal/rest/Push.kt]: The adaptar also... it'll return JSON 99% of the time 🤔

File: core/src/main/kotlin/chat/rocket/core/internal/rest/Push.kt (hunk 1 above)
Comment (leonardoaramaki) [reply core/src/main/kotlin/chat/rocket/core/internal/rest/Push.kt]: Independently of this content type being the most common or not we should definitely make this MediaType statically defined for best performance.

File: core/src/main/kotlin/chat/rocket/core/internal/rest/Push.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    val body = RequestBody.create(contentType, payloadBody)
+
+    val httpUrl = requestUrl(restUrl, "push.token").build()
+    val request = requestBuilder(httpUrl).post(body).build()
+
+    val type = Types.newParameterizedType(RestResult::class.java, PushToken::class.java)
+    handleRestCall<RestResult<PushToken>>(request, type).result()
+}
+
+suspend fun RocketChatClient.unregisterPushToken(token: String) {
```
Comment (luciofm) [inline core/src/main/kotlin/chat/rocket/core/internal/rest/Push.kt]: Use `withContext(CommonPool)` here also

File: core/src/main/kotlin/chat/rocket/core/internal/rest/Push.kt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import chat.rocket.core.RocketChatClient
+import chat.rocket.core.internal.RestResult
+import chat.rocket.core.internal.model.PushRegistrationPayload
+import chat.rocket.core.internal.model.PushUnregistrationPayload
+import chat.rocket.core.model.PushToken
+import com.squareup.moshi.Types
+import okhttp3.MediaType
+import okhttp3.RequestBody
+
+suspend fun RocketChatClient.registerPushToken(token: String) {
```
[the remark this answers is not in the document]
Comment (leonardoaramaki) [reply core/src/main/kotlin/chat/rocket/core/internal/rest/Push.kt]: IntelliJ couldn't find `withContext`. Anywhere on the code using it?

[the remark this answers is not in the document]
Comment (luciofm) [reply core/src/main/kotlin/chat/rocket/core/internal/rest/Push.kt]: Do a rebase/merge with develop

**Pull request merged**